### ISSUE TITLE
shows all the vim scroll history when switching tab

### ISSUE THREAD

**Issue by foisonocean** [austinjones/tab-rs#333]
When switch to a tab which uses vim, it will shows all the vim scroll history in that tab.
This is pretty annoying especially when use vim for a long time.

**Comment by TheWithz** [comment]
This happens to me fairly often.

**Comment by austinjones** [comment]
Yeah, I agree.  I've been thinking about some algorithms that would trim down the scrollback buffer.  But a few things I have tried in the past weren't fast enough.  What makes it a challenge is that data from the beginning of a session may be necessary to reconstruct it.  Apps set up a lot of terminal state on setup.  To actually trim down the size, it's necessary to remove content from the middle of the buffer.

So I've been thinking that tab would separate the scrollback history into 'layers', which could be independently pruned.  There would be a layer for stdout, and a layer for common types of terminal state.  It should be very fast to update/prune, and fast enough to reconstruct when a session is reconnected.

And it should be compatible with 'unknown' ANSI codes, as those would be still echoed.

**Comment by VoidNoire** [comment]
This issue seems to be related to another issue that causes my alacritty terminal to crash when I've been editing files on it with kakoune for a while and I switch back to that tab.
